src/Spice86.Core/Emulator/Memory/MainMemory.cs

```diff
@@ -0,0 +1,62 @@
+namespace Spice86.Core.Emulator.Memory; 
+
+using Spice86.Core.Emulator.VM;
+
+/// <summary>
+/// Represents the main memory of the IBM PC.
+/// Size must be at least 1 MB.
+/// </summary>
+public class MainMemory : Memory {
+    /// <summary>
+    /// Size of conventional memory in bytes.
+    /// </summary>
+    public const uint ConvMemorySize = 1024 * 1024;
+
+    private readonly Machine _machine;
+    
+    public MainMemory(Machine machine, uint sizeInKb) : base(sizeInKb) {
+        _machine = machine;
+        if (sizeInKb * 1024 < ConvMemorySize) {
+            throw new ArgumentException("Memory size must be at least 1 MB.");
+        }
+    }
+    
+    public override void SetUint32(uint address, uint value) {
+        if (_machine.Ems?.TryWriteMappedPageData(address, value) is not true) {
```

Comment:
Can we make this both more generic and performant?
Maybe by having a different Memory class per "area"? See https://www.zdnet.com/a/img/2014/10/03/3cfdb210-4b35-11e4-b6a0-d4ae52e95e57/memmap.jpg
Something like `_memoryBlocks[address >> 16].SetUint32(address, value)` where _memoryBlocks can be mapped by different classes?